libraries/ts/src/token/tokenAmount.ts

```diff
@@ -27,7 +27,7 @@ export class TokenAmount {
     this.lamports = lamports
     this.decimals = decimals
     this.tokens = TokenAmount.tokenAmount(lamports, decimals)
-    this.uiTokens = this.tokens.toLocaleString("fullwide", { useGrouping: true }) //to prevent scientific numbers. Grouping adds commas
+    this.uiTokens = this.tokens.toLocaleString("fullwide", { useGrouping: true, minimumFractionDigits: decimals }) //to prevent scientific numbers. Grouping adds commas
```

Comment:
I think you want to avoid trailing zeros. `maximumFractionDigits` may be better here?
```js
0.001.toLocaleString("fullwide", { useGrouping: true, minimumFractionDigits: 6 })
> '0.001000'
0.001.toLocaleString("fullwide", { useGrouping: true, maximumFractionDigits: 6 })
> '0.001'

```

```suggestion
    this.uiTokens = this.tokens.toLocaleString("fullwide", { useGrouping: true, maximumFractionDigits: decimals }) //to prevent scientific numbers. Grouping adds commas
```